Source/WPE/Source/Input/LibinputServer.cpp

```diff
@@ -204,6 +230,99 @@ void LibinputServer::dispatchKeyboardEvent(const Input::KeyboardEvent::Raw& even
     m_client->handleKeyboardEvent({ event.time, std::get<0>(result), std::get<1>(result), !!event.state, std::get<2>(result) });
 }
 
+
+void LibinputServer::handleTouchEvent(struct libinput_event *event, Input::TouchEvent::Type type, bool has_position)
+{
+    auto* touchEvent = libinput_event_get_touch_event(event);
+    uint32_t time = libinput_event_touch_get_time(touchEvent);
+    int id = libinput_event_touch_get_slot(touchEvent);
+    auto& targetPoint = m_touchEvents[id];
+    int32_t x, y;
+    
+    if (has_position) {
+      x = libinput_event_touch_get_x_transformed(touchEvent, m_touchSize.first);
+      y = libinput_event_touch_get_y_transformed(touchEvent, m_touchSize.second);
+    } else {
+      x = targetPoint.x;
+      y = targetPoint.y;
+    }
+    targetPoint = Input::TouchEvent::Raw{ type, time, id, x, y };
+    
+    m_client->handleTouchEvent({
+      m_touchEvents,
+      type,
+      id,
+      time
+    });
+    
+    if (type == Input::TouchEvent::Down) {
+        m_touchFingers++;
+        if (m_touchFingers == 1) {
+            m_touchClickId = id;
+            m_touchStartX = x;
+            m_touchStartY = y;
+            m_touchStartTime = time;
+        } else {
+           m_touchClickId = -1;
+        }
+    }
+    else if (type == Input::TouchEvent::Motion) {
+        if (id == m_touchClickId &&
+           (abs(m_touchStartX - x) > 20 || abs(m_touchStartY - y) > 20)
+        ){
+           m_touchClickId = -1;
+        }
+    }
+    else if (type == Input::TouchEvent::Up) {
+        if (id == m_touchClickId && time - m_touchStartTime < 2000) 
+        {
+            simulateClickEvent(time, (m_touchStartX + x) >> 1, (m_touchStartY + y) >> 1);
+            m_touchClickId = -1;
+        }
+        m_touchFingers--;
+        targetPoint = Input::TouchEvent::Raw{ Input::TouchEvent::Null, 0, -1, -1, -1 };
+    }
+}
+
+void LibinputServer::simulateClickEvent(uint32_t time, int32_t x, int32_t y) 
```

Comment:
I see. If you can provide a simple HTML sample (or just link to something that already exists) that showcases the problem I can help out.